contracts/escrow/dapps/CompoundFacet.sol

```diff
@@ -55,17 +56,38 @@ contract CompoundFacet is PausableMods, DappMods {
     ) public paused("", false) onlyBorrower(loanID) {
         ICErc20 cToken = AssetCTokenLib.get(tokenAddress);
 
-        LibEscrow.e(loanID).setTokenAllowance(tokenAddress, address(cToken));
-
-        bytes memory result = LibEscrow.e(loanID).callDapp(
-            address(cToken),
-            abi.encodeWithSelector(ICErc20.mint.selector, amount)
-        );
-
-        require(
-            abi.decode(result, (uint256)) == LibCompound.NO_ERROR,
-            "Teller: compound deposit error"
-        );
+        if (
+            keccak256(abi.encodePacked(cToken.symbol())) ==
+            keccak256(abi.encodePacked("cETH"))
+        ) {
+            // Unwrap WETH to deposit into cETH
+            IWETH(tokenAddress).withdraw(amount);
+
+            bytes memory result = LibEscrow.e(loanID).callDapp(
+                address(cToken),
+                abi.encodeWithSelector(ICErc20.mint.selector, amount)
+            );
+
+            require(
+                abi.decode(result, (uint256)) == LibCompound.NO_ERROR,
+                "Teller: compound deposit error"
+            );
+        } else {
+            LibEscrow.e(loanID).setTokenAllowance(
+                tokenAddress,
+                address(cToken)
+            );
+
+            bytes memory result = LibEscrow.e(loanID).callDapp(
+                address(cToken),
+                abi.encodeWithSelector(ICErc20.mint.selector, amount)
+            );
+
+            require(
+                abi.decode(result, (uint256)) == LibCompound.NO_ERROR,
+                "Teller: compound deposit error"
+            );
```

Comment:
since there are 2 of the same `require` check here lets save `result` as a value outside the `if` statement and the do the `require` after the `if`